plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/editor/context/project/ProjectContextProvider.kt

```diff
@@ -227,6 +227,7 @@ class ProjectContextProvider(val project: Project, private val encoderServer: En
         val encrypted = encoderServer.encrypt(payloadJson)
         with(url.openConnection() as HttpURLConnection) {
             setConnectionProperties(this)
+            setConnectionTimeout(this)
```

Comment:
i believe there was already a request to move to a different HTTP client, but even in the java stdlib, HttpClient is preferred over URLConnection